src/System.Private.CoreLib/shared/System/DefaultBinder.cs

```diff
@@ -1204,6 +1204,63 @@ private static bool CreateParamOrder(int[] paramOrder, ParameterInfo[] pars, str
             return true;
         }
 
+        // CanChangePrimitive
+        // This will determine if the source can be converted to the target type
+        private static bool CanChangePrimitive(Type source, Type target)
+        {
+            Primitives widerCodes = s_primitiveConversions[(int)(Type.GetTypeCode(source))];
```

Comment:
Are you sure this won't result in array out of bounds for some obscure types? (`ref int`, `int *`, etc.)

Reply:
As far as I understand `Type.GetTypeCode` return values in range 0..18
e.g.
```
Type.GetTypeCode(typeof(int*));
Type.GetTypeCode(typeof(IntPtr));
```
it returns TypeCode.Object (1).

Reply:
Thanks for checking. I hoped it would be the case.